### PR TITLE
Refactor/web3 plugin to provider

### PR DESCRIPTION
**Pull request by agualis**
# Description

 **Main changes:** 
- Refactor old web3 plugin into wallets provider
- Added tests for the new provider (80% coverage) 

**Extras:**
This was a critical global piece so it impacted many parts of the system: 
- Added and improved existing tests. 
- Refactor some components to <script setup>
- Testing tools improvements and cleanup

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/providers/createProviderComponent.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { defineComponent } from 'vue';
+
+export function createProviderComponent(provideCb: () => any) {
+  return defineComponent({
+    setup() {
+      provideCb();
```
Comment (garethfuller) [inline src/providers/createProviderComponent.ts:6]: I don't intuitively know what the 'Cb' stands for in this function name. I'd just rename it `provider`?
```suggestion
      provider();
```

File: src/services/web3/transactions/concerns/contract.concern.ts [hunk 2]
```diff
@@ -54,7 +66,9 @@ export class ContractConcern extends TransactionConcern {
       ]);
 
       trackGoal(Goals.ContractTransactionSubmitted);
-      return await contractWithSigner[action](...params, txOptions);
+
+      const result = await contractWithSigner[action](...params, txOptions);
+      return result;
```
Comment (garethfuller) [inline src/services/web3/transactions/concerns/contract.concern.ts:71]: Why break this onto two lines?

File: src/dependencies/wallets/default-mocks.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { initWalletconnectConnectorForTesting } from '@/dependencies/wallets/walletconnect';
+import { initWalletLinkConnectorForTesting } from '@/dependencies/wallets/walletlink';
+import { initSafeConnectorForTesting } from '@/dependencies/wallets/safe';
+import { initMetamaskConnectorForTesting } from './metamask';
+import {
+  WalletLinkConnectorMock,
+  WalletConnectConnectorMock,
+  GnosisConnectorMock,
```
Comment (garethfuller) [inline src/dependencies/wallets/default-mocks.ts:8]: Might want to rename the mocks too?

File: src/providers/wallet.provider.ts [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+  'walletlink',
+] as Wallet[];
+
+export const WalletNameMap: Record<Wallet, string> = {
+  metamask: 'Metamask',
+  walletconnect: 'WalletConnect',
+  gnosis: 'Gnosis Safe',
+  walletlink: 'Coinbase Wallet',
+  tally: 'Tally',
+};
```
Comment (garethfuller) [inline src/providers/wallet.provider.ts:62]: Maybe we should change the name here too?

File: src/services/web3/transactions/concerns/contract.concern.ts (hunk 2 above)
Comment (agualis) [reply src/services/web3/transactions/concerns/contract.concern.ts:71]: Probably forgot to undo after debugging.

**Review by garethfuller — APPROVED** [review]
Tested with a safe wallet and seems to work as before.

**Pull request merged**